### PR TITLE
Bugfix: Opposed HLR Inpaint border issues

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
The outermost photosites were not HLR corrected so far leading to artefacts for the outermost single photosites.
The effect depended on the chosen demosaicer.

I came across this while investigating #13924

### CODE REVIEW

File: data/kernels/basic.cl [hunk 1]
```diff
@@ -313,15 +313,22 @@ static float _calc_refavg(
...
         const unsigned int filters,
         int row,
-        int col)
+        int col,
+        int maxrow,
+        int maxcol)
 {
   float mean[3] = { 0.0f, 0.0f, 0.0f };
   float sum[3] =  { 0.0f, 0.0f, 0.0f };
   float cnt[3]  = { 0.0f, 0.0f, 0.0f };
```
Comment (ralfbrown) [inline data/kernels/basic.cl:322]: For a separate PR: would it make sense to declare these variables as dt_aligned_pixel_t?

Comment (jenshannoschwalm) [reply data/kernels/basic.cl:322]: From my current understanding how CL compilers work i would say no. One of the tasks they do is to translate as much variables as possible to internal registers as memory is very restricted performance-wise. That's roughly what i got from reading OpenCL "best practice suggestions". But - never tested that personally.

Comment (jenshannoschwalm) [reply data/kernels/basic.cl:322]: BTW: Just one thing i wanted to mention; have been away for 10days and just saw an "endless stream" of commits related to SSE2 code on my phone. Now back - just one word: CONGRATS!

Comment (ralfbrown) [reply data/kernels/basic.cl:322]: Thanks!   One more PR coming this weekend with a rewrite of the pixel interpolation code (sneak preview: 20+% speedup and removal of the majority of the SSE code).  Need to triple-check that I haven't broken anything.

Didn't notice that I was commenting on a .cl file.  I suspect that the CL compiler wouldn't support alignment directives anyway.

**Comment by TurboGit** [comment]
This breaks 0095 and confirm that your fix is correct:

diff

**Review by TurboGit — APPROVED** [review]
Thanks!

**Comment by jenshannoschwalm** [comment]
Release note:
Fixing border issue in inpaint opposed highlights reconstruction

**Comment by TurboGit** [comment]
Thanks!

**Pull request merged**